packages/api-plugin-carts/src/util/updateCartFulfillmentGroups.js

```diff
@@ -1,61 +1,123 @@
 import Random from "@reactioncommerce/random";
+import ReactionError from "@reactioncommerce/reaction-error";
 
 /**
- * @summary Figures out which fulfillment group a cart item should initially be in
+ * @summary Figures out from which fulfillment groups the item should be removed
  * @param {Object[]} currentGroups The current cart fulfillment groups array
- * @param {String[]} supportedFulfillmentTypes Array of fulfillment types supported by the item
+ * @param {String} itemId Id of the item to be removed
+ * @param {String} shopId The ID of the shop that owns the item (product)
+ * @returns {Object[]|null} The groups array or null if no viable group
+ */
+function determineRemoveFromGroups(currentGroups, itemId, shopId) {
+  let removeFromGroups = currentGroups.map((group) => {
+    if (group.itemIds && group.itemIds.includes(itemId) && shopId === group.shopId) {
+      return group;
+    }
+    return null;
+  });
+  removeFromGroups = removeFromGroups.filter((group) => !!group); // remove nulls
+  // If the item is present in more that one Fulfillment group, we remove it from all(for backward compatibility)
+  return removeFromGroups || null;
```

Comment:
`|| null` is redundant. 

```suggestion
  return _.compact(removeFromGroups);
```